subsys/net/lib/sockets/sockets.c

```diff
@@ -131,6 +131,16 @@ int zsock_socket_internal(int family, int type, int proto)
 
 int z_impl_zsock_socket(int family, int type, int proto)
 {
+	if (proto == 0) {
```

Comment:
Nack. This should be done, per address-family, in an address family implementation handler.

Reply:
This support has nothing to do with address family. Only thing important here is the type and proto so no need to duplicate the code in individual socket handler.

Reply:
OMG, this is something. Do you try to research fixes you make, or just sprinkle some random code around?

`man socket`:

>       The protocol specifies a  particular  protocol  to  be  used  with  the
>       socket.  Normally only a single protocol exists to support a particular
>       socket type within a given protocol family, in which case protocol  can
>       be  specified  as  0.   However, it is possible that many protocols may
>       exist, in which case a particular protocol must be  specified  in  this
       manner. 

So, no need to spread handling of socket() parameters around the codebase. The code which implements socket() (and that's socket handler) knows best how to handle the params.

Reply:
> The code which implements socket() (and that's socket handler) knows best how to handle the params.

Yep, that is why the check is in socket().
See also https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_10_18
We can certainly add here checks for `family==AF_INET || family == AF_INET6 || family == AF_PACKET`

Reply:
> OMG, this is something. Do you try to research fixes you make, or just sprinkle some random code around?

BTW, there is no need to write these "witty" remarks in your comments, it just makes harder for you to get your message through. If you find a bug in the commit, just say so.

Reply:
> BTW, there is no need to write these "witty" remarks

I'm sorry, but there's a stream of a patches from you (great work!), and I have a concern that a solution in almost each of them wasn't researched thoroughly enough. We're at the stage where a) we have a complex system; b) it mostly works. Any careless changes may throw it back into havoc. And if not, just make harder to maintain and debug (it already is!)

Reply:
> Yep, that is why the check is in socket().

socket() is nothing but a dispatcher which call whatever socket handler ready to handle the input params: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/sockets/sockets.c#L132 . We don't do any param checks there, it makes no sense - there can be arbitrary socket families, types, protocols, and only a handler for a particular combo (such handlers being configurable!) knows how to handle those params.

> See also https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_10_18

And what we read there? "When a socket is created in the **Internet family** with a protocol value of zero ..."

Once again, next day, there can be AF_KABOOM, and only socket handler for that family may know what it means to have protocol == 0 for it.